src/lib.rs

```diff
@@ -51,6 +51,7 @@ pub async fn run_trin(
         listen_port: trin_config.discovery_port,
         no_stun: trin_config.no_stun,
         bootnodes: trin_config.bootnodes.clone(),
+        node_data_dir: Some(node_data_dir.clone()),
```

Comment:
Something feels a bit off about this. It doesn't make any sense that `node_data_dir` could be `None`. A trin node should never have a `None` data dir. It seems like you went with an option here to support `PortalnetConfig::default()`?

Imo, maybe the best way forward is just to rename this variable, maybe something like `enr_record_file`. Since `enr_record_file`, in theory, could be none. eg. every node/datadir that runs this code for the first time won't have an enr record file.

Reply:
Sounds good I will use enr_file_location, as enr (ethereum node record), already has record and just file could be it is loaded into memory or something, good catch